pkg/clusterapi/system.go

```diff
@@ -79,6 +79,15 @@ func (c *system) Run(ctx context.Context, installConfig *installconfig.InstallCo
 	ctx, cancel := context.WithCancel(ctx)
 	c.cancel = cancel
 
+	//TODO(padillon): should we add a context.WithTimeout here?
```

Comment:
I think it would be useful to have a `context.WithTimeout` here - otherwise I guess a not-working provisioning could risk to be stuck forever?

Reply:
The timeout should be handled outside in this case, from the caller